### PR TITLE
k8s: Add DescribeConfigMap and DeleteConfigMap APIs

### PR DESCRIPTION
**Pull request by scarlettperry**
### Description

PR adds 2 APIs: DescribeConfigMap and DeleteConfigMap
* unit tests added

### Testing Performed
```
k get cm -n envoy-staging
NAME                       DATA   AGE
configmap-staging-test-1   1      18s
configmap-staging-test-2   1      31h
```

1. Get a ConfigMap by name
```
curl -X POST 'http://localhost:8080/v1/k8s/describeConfigMap' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "name": "configmap-staging-test-1"
}'
```

Response:
```
{"configMap":{"cluster":"kind-clutch-local", "namespace":"envoy-staging", "name":"configmap-staging-test-1", "annotations":{}, "labels":{"app":"envoy"}}}
```

2. Not found
```
curl -X POST 'http://localhost:8080/v1/k8s/describeConfigMap' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "name": "abc"
}'
```

Response:
```
{"code":2, "message":"Unable to locate configMap", "details":[]}
```

3. Delete a ConfigMap by name
```
curl -X POST 'http://localhost:8080/v1/k8s/deleteConfigMap' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "name": "configmap-staging-test-1"
}'
```

```
k get cm -n envoy-staging
NAME                       DATA   AGE
configmap-staging-test-2   1      32h
```

### CODE REVIEW

File: api/k8s/v1/k8s.proto [hunk 1]
```diff
@@ -545,6 +561,41 @@ message ListConfigMapsResponse {
...
+    type_url : "clutch.k8s.v1.ConfigMap",
+    pattern : "{cluster}/{namespace}/{name}"
+  };
+
+  string clientset = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string cluster = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  string namespace = 3 [ (validate.rules).string = {min_bytes : 1} ];
+  string name = 4 [ (validate.rules).string = {min_bytes : 1} ];
+
+  map<string, string> labels = 5 [ (validate.rules).map.keys.string = {min_bytes : 1} ];
```
Comment (rithujohn191) [inline api/k8s/v1/k8s.proto:575]: Should we add annotations too?

File: backend/service/k8s/configmap.go [hunk 2]
```diff
@@ -2,12 +2,47 @@ package k8s
...
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(configMapList.Items) == 1 {
+		return protoForConfigMap(cs.Cluster(), &configMapList.Items[0]), nil
+	} else if len(configMapList.Items) > 1 {
+		return nil, fmt.Errorf("Located multiple configMaps")
```
Comment (rithujohn191) [inline backend/service/k8s/configmap.go:30]: Maybe say `"Located multiple configMaps with name %s", name`? 

Technically we should never hit this case cause kubernetes would never allow more than once configmap with the same name and we filter on the name in the list command

Comment (rithujohn191) [reply backend/service/k8s/configmap.go:30]: All the other resources have the same error message, so I guess its okay to leave it as is :) upto you :)

File: backend/service/k8s/configmap_test.go [hunk 3]
```diff
@@ -90,6 +103,53 @@ func TestListConfigMaps(t *testing.T) {
...
+		manager: &managerImpl{
+			clientsets: map[string]*ctxClientsetImpl{"testing-clientset": &ctxClientsetImpl{
+				Interface: cs,
+				namespace: "testing-namespace",
+				cluster:   "testing-cluster",
+			}},
+		},
+	}
+
+	configMap, err := s.DescribeConfigMap(context.Background(), "testing-clientset", "testing-cluster", "testing-namespace", "testing-configmap-name")
```
Comment (rithujohn191) [inline backend/service/k8s/configmap_test.go:121]: Could you also add another case to Describe a configmap that does not exist and ensure there is an error

File: api/k8s/v1/k8s.proto (hunk 1 above)
Comment (scarlettperry) [reply api/k8s/v1/k8s.proto:575]: sure! added in https://github.com/lyft/clutch/pull/930/commits/147f0d0b6cd71926d777622693c9b8e90e5b7349

File: backend/service/k8s/configmap.go (hunk 2 above)
Comment (scarlettperry) [reply backend/service/k8s/configmap.go]: added in https://github.com/lyft/clutch/pull/930/commits/147f0d0b6cd71926d777622693c9b8e90e5b7349

File: backend/service/k8s/configmap_test.go (hunk 3 above)
Comment (scarlettperry) [reply backend/service/k8s/configmap_test.go:126]: added in https://github.com/lyft/clutch/pull/930/commits/147f0d0b6cd71926d777622693c9b8e90e5b7349

TIL that the fakeclient doesn't filter by field selectors. was raised in https://github.com/kubernetes/client-go/issues/326 which linked me to other issues. so I followed the pattern I've seen in other tests for "not found"  https://github.com/lyft/clutch/blob/main/backend/service/k8s/pods_test.go#L129

**Review by rithujohn191 — APPROVED** [review]
👌

**Pull request merged**